scripts/python/phdf_diff.py

```diff
@@ -136,9 +167,50 @@ def compare(files, all=False, brief=True, quiet=False, one=False, tol=1.0e-12):
         otherLocations[idx] = f0.findIndexInOther(f1,idx)
     if not quiet: print(f0.TotalCells,'cells mapped')
 
+   
+    otherBlockIdx = list(f0.findBlockIdxInOther(f1,i) for i in range(f0.NumBlocks))
+
     for var in f0.Variables:
-        if var == 'Locations' or var == 'Info':
+
+        if var in ['Locations', 'VolumeLocations', 'Info', 'LogicalLocations', 'Levels']:
+            if check_meta_data and var != 'Info':
+                if var in ['Locations', 'VolumeLocations']:
+                    for key in f0.fid[var].keys():
+                        #Compare raw data of these variables
+                        val0 = f0.fid[var][key]
+                        val1 = f1.fid[var][key]
+
+                        #Sort val1 by otherBlockIdx
+                        val1 = val1[otherBlockIdx]
+
+                        # Compute norm error, check against tolerance
+                        errVal = np.abs(val0 - val1)
+                        errMag = np.linalg.norm(errVal)
+                        if(errMag > tol):
+                            no_meta_variables_diff = False
+                            if not quiet: print("")
+                            print(f'Metavariable {var}/{key} differs between {f0.file} and {f1.file}')
+                            if not quiet: print("")
+                        else:
+                            print('  %18s/%s: no diffs'%(var,key))
+                if var in ['LogicalLocations', 'Levels']:
+                    #Compare raw data of these variables
+                    val0 = f0.fid[var]
+                    val1 = f1.fid[var]
+
+                    #Sort val1 by otherBlockIdx
+                    val1 = val1[otherBlockIdx]
+
+                    #As integers, they should be identical
+                    if val0 != val0:
+                        no_meta_variables_diff = False
+                        if not quiet: print("")
+                        print(f'Metavariable {var} differs between {f0.file} and {f1.file}')
+                        if not quiet: print("")
+                    else:
+                        print('  %20s: no diffs'%var)
```

Comment:
Again I would recommend packaging this block of logic into a separate method, ... preference. 
for floating point message I would print the difference.